[DOCS] explain ysql java tests (#4575)
yugabyte/yugabyte-db: docs/content/latest/contribute/core-database/run-unit-tests.md

{{< note title="Note" >}}
The YB logs are contained in the output file now.
{{< /note >}}

### YSQL Java tests

YSQL java tests are in `java/yb-pgsql/src/test/java/org/yb/pgsql/`.  They can be run as:
```bash

> ./yb_build.sh --java-test TestPgTruncate

Reviewer: Actually, this is a shorthand that could possibly conflict with stuff in `java/yb-cql`, for example.  Use `--java-test org.yb.pgsql.TestPgTruncate`.

```

> Some of those tests, `TestPgRegress*` use the postgres regress test framework: `src/postgres/src/test/regress`.  

Reviewer: style: missing comma after `TestPgRegress*`.

They should each correspond to a schedule (e.g. `java/yb-pgsql/src/test/java/org/yb/pgsql/TestPgRegressArrays.java` references `src/postgres/src/test/regress/yb_arrays_schedule`) 
that is run by our modified version of `pg_regress`.

> Each schedule has a serial order of files to run.  For example, the `yb_arrays_schedule` will first run `build/latest/postgres_build/src/test/regress/sql/yb_pg_int8.sql` 

Reviewer: Because this has a chance of becoming outdated, explain why: the first `test:` line in `yb_arrays_schedule` is `test: yb_pg_int8`.

Author: I've commited all changes. Please quick review and I commit.

> and output to `build/latest/postgres_build/src/test/regress/results/yb_pg_int8.out`.  This will be compared with `build/latest/postgres_build/src/test/regress/expected/yb_pg_int8.out for pass/fail`.  

Reviewer: style: properly place backtick before " for pass/fail".

Note the `build/latest/postgres_build` prefix.  The source files (`src/postgres/src/test/regress/sql/foo.sql`) get copied there (`build/latest/postgres_build/src/test/regress/src/foo.sql`).  
If a build fails to pick up the changes and fails to copy them, you can remove `build/latest/postgres_build` to force a recopy or copy them yourself manually.

{{< tip title="Tips" >}}

   - If you want to quickly run specific sql files, you can create a dummy java file and dummy schedule with that one test in it.
   - Use the naming convention (some older files haven't adopted it yet, but should):
     - `src/postgres/src/test/regress/sql/foo.sql`: unchanged from original postgres code
     - `src/postgres/src/test/regress/sql/yb_foo.sql`: completely new file (for example, with new features)
     - `src/postgres/src/test/regress/sql/yb_pg_foo.sql`: modified version of original postgres foo.sql (e.g. for compatibility edits) 
     - The goal here is to reduce the difference between `foo.sql` and `yb_pg_foo.sql`, when possible.

> 

Reviewer: Add tip:
> When creating new `yb_pg_foo.{out,sql}` files and adding them to one of our schedules, sort them into the schedule using `src/postgres/src/test/regress/serial_schedule` as reference.  Schedules `parallel_schedule` and `serial_schedule` should be untouched as they are from original postgres code.

{{< note title="Note" >}}
The YB logs are contained in the output file now.
{{< /note >}}

### YSQL Java tests

YSQL java tests are in `java/yb-pgsql/src/test/java/org/yb/pgsql/`.  They can be run as:
```bash
./yb_build.sh --java-test TestPgTruncate
```
Some of those tests, `TestPgRegress*` use the postgres regress test framework: `src/postgres/src/test/regress`.  
They should each correspond to a schedule (e.g. `java/yb-pgsql/src/test/java/org/yb/pgsql/TestPgRegressArrays.java` references `src/postgres/src/test/regress/yb_arrays_schedule`) 

> that is run by our modified version of pg_regress.

Reviewer 2: ```suggestion
that is run by our modified version of `pg_regress`.
```

> Each schedule has a serial order of filenames to run.  For example, the `yb_arrays_schedule` will first run `build/latest/postgres_build/src/test/regress/sql/yb_pg_int8.sql` 

Reviewer 2: ```suggestion
Each schedule has a serial order of file names to run.  For example, the `yb_arrays_schedule` will first run `build/latest/postgres_build/src/test/regress/sql/yb_pg_int8.sql` 
```

and output to `build/latest/postgres_build/src/test/regress/results/yb_pg_int8.out`.  This will be compared with `build/latest/postgres_build/src/test/regress/expected/yb_pg_int8.out for pass/fail`.  
Note the `build/latest/postgres_build` prefix.  The source files (`src/postgres/src/test/regress/sql/foo.sql`) get copied there (`build/latest/postgres_build/src/test/regress/src/foo.sql`).  

> If build somehow doesn't pick up the changes and doesn't copy, you can remove `build/latest/postgres_build` to force recopy or copy them manually yourself.

Reviewer 2: ```suggestion
If a build fails to pick up the changes and fails to copy them, you can remove `build/latest/postgres_build` to force a recopy or copy them yourself manually.
```

{{< tip title="Tips" >}}

   - If you want to quickly run specific sql files, you can create a dummy java file and dummy schedule with that one test in it.

>    - We have a naming convention (some older files haven't adopted it yet but should):

Reviewer 2: ```suggestion
   - Use the naming convention (some older files haven't adopted it yet, but should):
```

     - `src/postgres/src/test/regress/sql/foo.sql`: unchanged from original postgres code

>      - `src/postgres/src/test/regress/sql/yb_foo.sql`: completely new file (e.g. for new features)

Reviewer 2: ```suggestion
     - `src/postgres/src/test/regress/sql/yb_foo.sql`: completely new file (for example, with new features)
```

     - `src/postgres/src/test/regress/sql/yb_pg_foo.sql`: modified version of original postgres foo.sql (e.g. for compatibility edits) 

>      - the aim here is to reduce diff between `foo.sql` and `yb_pg_foo.sql` when possible

Reviewer 2: ```suggestion
     - The goal here is to reduce the difference between `foo.sql` and `yb_pg_foo.sql`, when possible.
```
